app/[locale]/(interactive)/4-6/page.tsx

```diff
@@ -0,0 +1,145 @@
+"use client";
+import NextButton from "@/components/ui/nextButton";
+import { useNameStorkeStore } from "@/stores/NameStroke.store";
+import Link from "next/link";
+import { getStroke } from "perfect-freehand";
+import React, { useRef } from "react";
+
+type Props = {};
+
+const Page = (props: Props) => {
+  const nameStorke = useNameStorkeStore((state) => state.nameStorke);
+  const updateNameStorke = useNameStorkeStore(
+    (state) => state.updateNameStorke,
+  );
+
+  function getSvgPathFromStroke(stroke: number[][]) {
+    if (!stroke.length) return "";
+    const d = stroke.reduce(
+      (acc: any[], [x0, y0]: any, i: number, arr: string | any[]) => {
+        const [x1, y1] = arr[(i + 1) % arr.length];
+        acc.push(x0, y0, (x0 + x1) / 2, (y0 + y1) / 2);
+        return acc;
+      },
+      ["M", ...stroke[0], "Q"],
+    );
+    d.push("Z");
+    return d.join(" ");
+  }
+
+  const [hasWrite, setWrite] = React.useState<boolean>(false);
+  const currentStroke = useRef<number[][]>([]);
+  const svgRef = useRef();
+
+  function handlePointerDown(e: React.PointerEvent<SVGSVGElement>) {
+    setWrite(true);
+    e.currentTarget.setPointerCapture(e.pointerId);
+    const pos = svgRef.current.getBoundingClientRect();
+    currentStroke.current = [
+      [e.pageX - pos.left, e.pageY - pos.top, e.pressure],
+    ];
+  }
+
+  function handlePointerMove(e: React.PointerEvent<SVGSVGElement>) {
+    if (e.buttons !== 1) return;
+    const pos = svgRef.current.getBoundingClientRect();
+    currentStroke.current.push([
+      e.pageX - pos.left,
+      e.pageY - pos.top,
+      e.pressure,
+    ]);
+    updateNameStorke([...nameStorke]);
+  }
+
+  function handlePointerUp() {
+    // new function to handle new independent stroke
+    updateNameStorke([...nameStorke, currentStroke.current]);
+    currentStroke.current = [];
+  }
+
+  function clearStrokes() {
+    setWrite(false);
+    updateNameStorke([]);
+  }
+
+  const renderedStrokes = nameStorke.map((stroke, index) => {
+    const pathData = getSvgPathFromStroke(
+      getStroke(stroke, {
+        size: 8,
+        thinning: 0.5,
+        smoothing: 0.5,
+        streamline: 0.5,
+      }),
+    );
+    return <path key={index} d={pathData} />;
+  });
```

Comment:
looks like there are multiple svg-related operations which is duplicated among the codebase. Please tidy this up